lib/core/trade_monitor.dart

```diff
@@ -0,0 +1,171 @@
+        return TrocadorExchangeProvider();
+      case ExchangeProviderDescription.exolix:
+        return ExolixExchangeProvider();
+      case ExchangeProviderDescription.thorChain:
+        return ThorChainExchangeProvider(tradesStore: trades);
+      case ExchangeProviderDescription.swapTrade:
+        return SwapTradeExchangeProvider();
+      case ExchangeProviderDescription.letsExchange:
+        return LetsExchangeExchangeProvider();
+      case ExchangeProviderDescription.stealthEx:
+        return StealthExExchangeProvider();
+      case ExchangeProviderDescription.chainflip:
+        return ChainflipExchangeProvider(tradesStore: trades);
+      case ExchangeProviderDescription.xoSwap:
+        return XOSwapExchangeProvider();
+    }
+    return null;
+  }
+
+  void monitorActiveTrades(String walletId) {
+    final now = DateTime.now();
+    final trades = tradesStore.trades;
+
+    for (final item in trades) {
+      final trade = item.trade;
+
+      if (trade.walletId != walletId) continue;
+
+      final createdAt = trade.createdAt;
+      if (createdAt == null) continue;
+
+      if (now.difference(createdAt).inHours > 24) continue;
+
+      if (_isFinalState(trade.state)) continue;
+
+      if (!_tradeTimers.containsKey(trade.id)) {
+        printV('Starting trade monitoring for ${trade.id}');
+        _startTradeMonitoring(trade);
+      }
+    }
+  }
+
+  void _startTradeMonitoring(Trade trade) {
+    if (_tradeTimers.containsKey(trade.id)) return;
+
+    final provider = _getProviderByDescription(trade.provider);
+    if (provider == null) {
+      printV('No provider found for trade ${trade.id}');
+      return;
+    }
+
+    _checkTradeStatus(trade);
+
+    final timer = Timer.periodic(Duration(minutes: _tradeCheckInterval), (_) {
+      _checkTradeStatus(trade);
+    });
+
+    _tradeTimers[trade.id] = timer;
+  }
+
+  Future<void> _checkTradeStatus(Trade trade) async {
+    printV('Checking trade status for ${trade.id}');
+    final provider = _getProviderByDescription(trade.provider);
+    if (provider == null) {
+      printV('No provider found for trade ${trade.id}');
+      return;
+    }
+
+    final exchangeApiMode = settingsStore.exchangeStatus;
+
+    if (exchangeApiMode == ExchangeApiMode.disabled) {
+      printV('Exchange API mode is disabled');
+      return;
+    }
+
+    if (exchangeApiMode == ExchangeApiMode.torOnly && !provider.supportsOnionAddress) {
+      printV('Skipping ${trade.provider}, no TOR support');
+      return;
+    }
+
+    try {
+      final updated = await provider.findTradeById(id: trade.id);
+      trade
+        ..stateRaw = updated.state.raw
+        ..receiveAmount = updated.receiveAmount
+        ..outputTransaction = updated.outputTransaction;
+      printV('Trade ${trade.id} updated: ${trade.state}');
+      await trade.save();
+
+      if (_isFinalState(updated.state)) {
+        printV('Trade ${trade.id} is in final state');
+        _cancelTradeTimer(trade.id);
+      }
```

Comment:
here you only cancel it when it gets to a final state, but we should also cancel if it exceeds 24 hours.

Reply:
Alright, I'll add that check